src/main/java/pl/javastart/streamsexercise/PaymentService.java

```diff
@@ -19,70 +21,117 @@ class PaymentService {
+                .stream()
+                .filter(payment -> payment.getPaymentDate().getMonth().equals(zonedDateTime.getMonth()))
+                .collect(Collectors.toList());
     }
 
     /*
     Znajdź i zwróć płatności dla wskazanego miesiąca
      */
     List<Payment> findPaymentsForGivenMonth(YearMonth yearMonth) {
-        throw new RuntimeException("Not implemented");
+        return paymentRepository.findAll()
+                .stream()
+                .filter(payment -> payment.getPaymentDate().getMonth().equals(yearMonth.getMonth()))
+                .collect(Collectors.toList());
     }
 
     /*
     Znajdź i zwróć płatności dla ostatnich X dzni
      */
     List<Payment> findPaymentsForGivenLastDays(int days) {
+//        return paymentRepository.findAll()
+//                .stream()
+//                .filter(payment -> payment.getPaymentDate())
+//                .collect(Collectors.toList());
         throw new RuntimeException("Not implemented");
     }
 
     /*
     Znajdź i zwróć płatności z jednym elementem
      */
     Set<Payment> findPaymentsWithOnePaymentItem() {
-        throw new RuntimeException("Not implemented");
+        return paymentRepository.findAll().stream()
+                .filter(payment -> payment.getPaymentItems().size() == 1)
+                .collect(Collectors.toSet());
     }
 
     /*
     Znajdź i zwróć nazwy produktów sprzedanych w aktualnym miesiącu
      */
     Set<String> findProductsSoldInCurrentMonth() {
-        throw new RuntimeException("Not implemented");
+        ZonedDateTime zonedDateTime = dateTimeProvider.zonedDateTimeNow();
+        return paymentRepository.findAll()
+                .stream()
+                .filter(payment -> payment.getPaymentDate().getMonth().equals(zonedDateTime.getMonth()))
+                .map(Payment::getPaymentItems)
+                .flatMap(Collection::stream)
+                .map(PaymentItem::getName)
+                .collect(Collectors.toSet());
     }
 
     /*
     Policz i zwróć sumę sprzedaży dla wskazanego miesiąca
      */
     BigDecimal sumTotalForGivenMonth(YearMonth yearMonth) {
+//        return paymentRepository.findAll()
+//                .stream()
+//                .map(Payment::getPaymentItems)
+//                .flatMap(Collection::stream)
+//                .map(PaymentItem::getFinalPrice)
+//                .reduce(BigDecimal::add).get()
         throw new RuntimeException("Not implemented");
     }
 
     /*
     Policz i zwróć sumę przeyznanaych rabatów dla wskazanego miesiąca
      */
     BigDecimal sumDiscountForGivenMonth(YearMonth yearMonth) {
+//        return paymentRepository.findAll().stream()
+//                .filter(payment -> payment.getPaymentDate().getMonth().equals(yearMonth.getMonth()))
+//                .map(Payment::getPaymentItems)
+//                .flatMap(Collection::stream)
+//                .filter(paymentItem ->
+//                .collect(Collectors.toList();
```

Comment:
Wykorzystaj findPaymentsForGivenMonth, dalej mapowanie i flatMap masz zrobione ok. Filtrowanie nie będzie potrzebne, ale zamiast tego będziesz musiała zrobić kolejne mapowanie, w którym obliczysz różnicę regularPrice-finalPrice i na końcu to zsumujesz